### PR TITLE
Add LDA example to Microsoft.ML.Samples

### PR DESCRIPTION
**Pull request by abgoswam**
- Adding a LDA example to Microsoft.ML.Samples . This was a pending comment on #1410

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            string ldaFeatures = "LdaFeatures";
+            var pipeline = ml.Transforms.Text.ProduceWordBags("Review").
+                Append(ml.Transforms.Text.LatentDirichletAllocation("Review", ldaFeatures, numTopic:3));
+
+            // The transformed data
+            var transformer = pipeline.Fit(trainData);
+            var transformed_data = transformer.Transform(trainData);
+
+            // Small helper to print the text inside the columns, in the console. 
+            Action<string, IEnumerable<VBuffer<float>>> printHelper = (columnName, column) =>
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs]: >printHelper [](start = 56, length = 11)

you don't have to do this, if you'll use it only once.

File: docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            // Review                               
+            // "animals birds cats dogs fish horse" 
+            // "horse birds house fish duck cats"   
+            // "car truck driver bus pickup"       
+            // "car truck driver bus pickup horse"
+
+            // A pipeline for featurizing the "Review" column
+            string ldaFeatures = "LdaFeatures";
+            var pipeline = ml.Transforms.Text.ProduceWordBags("Review").
+                Append(ml.Transforms.Text.LatentDirichletAllocation("Review", ldaFeatures, numTopic:3));
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs]: >LatentDirichletAllocation [](start = 42, length = 25)

just asking: besides numTopic, is there other params that the user might tune often?

**Comment by sfilipi** [comment]
>         public static LatentDirichletAllocationEstimator LatentDirichletAllocation(this TransformsCatalog.TextTransforms catalog, params LatentDirichletAllocationTransformer.ColumnInfo[] columns)

maybe add the same sample one more time, and adapt to use this one, so this gets its example as well? 

---
Refers to: src/Microsoft.ML.Transforms/Text/TextCatalog.cs:540 in cbd8ad3. [](commit_id = cbd8ad36a85e50a8b6859d4a3d72f6f2763bb8f0, deletion_comment = False)

File: docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            //
+            // Review                               
+            // "animals birds cats dogs fish horse" 
+            // "horse birds house fish duck cats"   
+            // "car truck driver bus pickup"       
+            // "car truck driver bus pickup horse"
+
+            // A pipeline for featurizing the "Review" column
+            string ldaFeatures = "LdaFeatures";
+            var pipeline = ml.Transforms.Text.ProduceWordBags("Review").
```
Comment (Ivanidzo4ka) [inline docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs]: >"Review" [](start = 62, length = 8)

nameof(SamplesUtils.DatasetUtils.SampleTopicsData.Review) ?

File: docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs (hunk 2 above)
Comment (abgoswam) [reply docs/samples/Microsoft.ML.Samples/Dynamic/LdaTransform.cs]: i doubt users tune other parameters besides numTopic

---
In reply to: [237660759](https://github.com/dotnet/machinelearning/pull/1782#discussion_r237660759) [](ancestors = 237660759)

**Comment by abgoswam** [comment]
>         public static LatentDirichletAllocationEstimator LatentDirichletAllocation(this TransformsCatalog.TextTransforms catalog, params LatentDirichletAllocationTransformer.ColumnInfo[] columns)

none of the examples within Microsoft.Ml.Samples  showcase this paradigm .. Is it to keep things simple ? 

I am skipping this for now.

---
In reply to: [442996256](https://github.com/dotnet/machinelearning/pull/1782#issuecomment-442996256) [](ancestors = 442996256)

---
Refers to: src/Microsoft.ML.Transforms/Text/TextCatalog.cs:540 in cbd8ad3. [](commit_id = cbd8ad36a85e50a8b6859d4a3d72f6f2763bb8f0, deletion_comment = False)

**Pull request merged**